### PR TITLE
Teapot test

### PR DESCRIPTION
**Pull request by hell0w0rld2**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Associated Issue(s)
Issue: <#X or issue-XX > [REMOVE THIS WHEN DONE Describe the issue you're working on here so it ties to the issue and Kanban (IMPORTANT FOR TRACKING). Make sure to create an issue if one doesn't already exist! ]

## Steps to Test
[ Describe how to run your AUTOMATED TESTS and any manual tests that may be need to prove your fix/feature REMOVE THIS WHEN DONE]
1. [First Step]
2. [Second Step]
3. [and so on...]

## Changelog

Working mostly complete smoke test for teapot. Hopper code needs to be added. Also adds compatibility for teapot and other changes for the code to work.

## What?

Working teapot smoke test

## Why?

Allows us to be able to test to see if all of the components of the robot are working before we go into a match.

### CODE REVIEW

**Review by ManiacMoustache — APPROVED** [review]
it make work good

File: autonomous/smokeTest.py [hunk 1]
```diff
@@ -85,20 +86,23 @@ def runShooterMotors(self):
...
     def calibrateTurret(self):
         """Calibrates the turret's deadzones and checks to see if the turret motor is working"""
         self.toDo = "Check to see if the turret is moving and that the deadzones are calibrated"
         self.shooterMotors.stopShooter()
-        self.next_state = "colorSensorCheck"
+        self.next_state("colorSensorCheck")
 
     @state
     def colorSensorCheck(self):
         """Has the user put up a red ball to the color sensor. Will not move on until the ball is red."""
```
Comment (ChrisHirsch) [inline autonomous/smokeTest.py:103]: Can we add the exact same test for a blue ball too? Basically make them pass red and then make them pass blue?

File: components/Input/colorSensor.py [hunk 2]
```diff
@@ -52,4 +53,5 @@ def displayColor(self):
         return self.colorMatched
 
     def execute(self):
+        self.displayColor()
```
Comment (ChrisHirsch) [inline components/Input/colorSensor.py:56]: not needed?

File: configs/teapot.yml [hunk 3]
```diff
@@ -19,7 +19,7 @@ driveTrain:
     rightMotor:
       channel: 30
       type: "CANTalonFX"
-      inverted: False
+      inverted: True
```
Comment (ChrisHirsch) [inline configs/teapot.yml:22]: This fixes the reversed motors right?

**Review by ChrisHirsch — COMMENTED** [review]
See if we can make these changes and I'll be more than happy to approve. If not, just create issues (around adding another ball check, etc) and we will get this approved

File: configs/teapot.yml (hunk 3 above)
Comment (hell0w0rld2) [reply configs/teapot.yml:22]: Yes

File: components/Input/colorSensor.py (hunk 2 above)
Comment (tuffinmuffin) [reply components/Input/colorSensor.py:56]: think feedback handles it

**Review by Leaf-life — APPROVED** [review]
good

**Review by tuffinmuffin — APPROVED** [review]

**Pull request merged**